en/mavgen_c/message_signing_c.md

```diff
@@ -0,0 +1,132 @@
+# C Message Signing (mavgen)
+
+One of the key features of [MAVLink 2](../guide/mavlink_2.md) is support for [Message Signing](../guide/message_signing.md) (authentication).
+
+The C libraries generated by *mavgen* provide almost everything needed to support signing in your MAVLink system.
+You will need to add some code to:
+
+* Handle the [SETUP_SIGNING](../messages/common.md#SETUP_SIGNING) message.
+* Setup and teardown signing on a link.
+* Save and load the secret key and timestamp in persistent storage
+* Implement a callback to define which (if any) unsigned messages will be accepted.
+
+
+## Handling SETUP_SIGNING
+
+The [SETUP_SIGNING](../messages/common.md#SETUP_SIGNING) message is the mechanism for a GCS to setup a signing key on a *MAVLink 2* device. 
+It takes a 32 byte secret key and an initial timestamp. 
+The method of generating the 32 byte secret key is up to the GCS implementation, although it is suggested that all GCS implementations should support the use of a sha256 hash of a user provided passphrase.
+
+## Handling Timestamps
```

Comment:
@LorenzMeier It is clear what the framework does for you, but not how you should do the bits you are expected to. 

Questions
- How is a system using this library supposed to get hold of the current timestamp for saving?
- How is the system supposed to "inject" a restored timestamp back into the back into the library? For example, on getting GPS signal.